interpreter/golabels/golabels.go

```diff
@@ -59,7 +62,14 @@ func Loader(_ interpreter.EbpfHandler, info *interpreter.LoaderInfo) (interprete
 
 	offsets := getOffsets(goVersion)
 	tlsOffset, err := extractTLSGOffset(file)
-	if err != nil {
+	switch err {
+	case libpf.ErrSymbolNotFound:
+		return nil, fmt.Errorf("failed to lookup symbol in %s: %v", info.FileName(), err)
+	case errDecodeSymbol:
+		log.Warnf("Failed to decode symbol in %s for Go label extraction", info.FileName())
```

Comment:
This lost the symbol name we tried to decode. So the text is slightly confusing for me. Would it be possible to return an error with the symbol name, and use `errors.Is` to detect this, and include `err` in the warning?

Perhaps convert the whole `switch` to `if err != nil` block and then use further `errors.Is`.